frame/composable-traits/src/staking/mod.rs

```diff
@@ -0,0 +1,147 @@
+use crate::staking::{
+	lock::{Lock, LockConfig},
+	rewards::RewardConfig,
+};
+use codec::{Decode, Encode};
+use composable_support::collections::vec::bounded::BiBoundedVec;
+
+use core::fmt::Debug;
+use frame_support::dispatch::DispatchResult;
+use scale_info::TypeInfo;
+use sp_runtime::{DispatchError, Perbill, Permill};
+
+pub mod lock;
+pub mod math;
+pub mod nft;
+pub mod rewards;
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct StakeConfig<DurationPresets, RewardsRate> {
+	pub lock: Option<LockConfig<DurationPresets>>,
+	pub reward: Option<RewardConfig<RewardsRate>>,
+}
+
+/// staking typed fNFT, usually can be mapped to raw fNFT storage type
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Encode, Decode, TypeInfo)]
+pub struct Stake<Balance, Rewards> {
+	/// The original stake this NFT was minted for or updated NFT with increased stake amount.
+	pub real_stake: Balance,
+	/// List of reward asset/pending rewards.
+	pub rewards: Rewards,
+	pub lock: Option<Lock>,
+	/// The reward multiplier.
+	pub reward_multiplier: Perbill,
+}
+
+/// implemented by instances which know their share of something bigger
+pub trait Shares {
+	type Balance;
+	fn shares(&self) -> Self::Balance;
+}
+
+pub trait ProtocolStaking {
+	type AccountId;
+	type AssetId;
+	type Balance;
+	type InstanceId;
+	type PoolId;
+
+	/// adds reward to common pool share
+	fn accumulate_reward(
+		pool: &Self::PoolId,
+		reward_currency: Self::AssetId,
+		reward_increment: Self::Balance,
+	) -> DispatchResult;
+}
+
+/// Interface for protocol staking.
+pub trait Staking {
+	type AccountId;
+	type PoolId;
+	type Balance;
+	type InstanceId;
+
+	/// Stake an amount of protocol asset. A new NFT representing the user position will be
+	/// minted.
+	///
+	/// Arguments
+	///
+	/// * `asset` the protocol asset to stake.
+	/// * `from` the account to transfer the stake from.
+	/// * `amount` the amount to stake.
+	/// * `duration` the staking duration (must be one of the predefined presets). Unstaking before
+	///   the end trigger the unstake penalty.
+	/// * `keep_alive` whether to keep the `from` account alive or not while transferring the stake.
+	/// * `config_index` - reference to one of  staking configs for `pool_id`
+	fn create_share(
+		who: &Self::AccountId,
+		pool_id: &Self::PoolId,
+		config_index: u8,
```

Comment:
`PoolId` already identifies the required config
```suggestion
```

Reply:
it is per position config, not per pool, please check mockups attached to clickup